libcudacxx/include/cuda/std/inplace_vector

```diff
@@ -0,0 +1,1808 @@
+      this->__uninitialized_fill(__end, __end + __diff, __value);
+    }
+  }
+
+  _LIBCUDACXX_INLINE_VISIBILITY static constexpr void reserve(const size_type __count)
+  {
+    if (_Capacity < __count)
+    {
+      _CUDA_VSTD::__throw_bad_alloc();
+    }
+  }
+  _LIBCUDACXX_INLINE_VISIBILITY static constexpr void shrink_to_fit() noexcept {}
+
+  _LIBCUDACXX_TEMPLATE(class _Tp2 = _Tp)
+  _LIBCUDACXX_REQUIRES(_CCCL_TRAIT(is_swappable, _Tp2) _LIBCUDACXX_AND _CCCL_TRAIT(is_move_constructible, _Tp2))
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr void swap(inplace_vector& __other) noexcept(
+    _Capacity == 0 || (_CCCL_TRAIT(is_nothrow_swappable, _Tp2) && _CCCL_TRAIT(is_nothrow_move_constructible, _Tp2)))
+  {
+    _CCCL_IF_CONSTEXPR (_Capacity == 0)
+    {
+      return;
+    }
+
+    if (this->size() < __other.size())
+    {
+      const auto __new_mid = _CUDA_VSTD::swap_ranges(this->begin(), this->end(), __other.begin());
+      this->__uninitialized_move(__new_mid, __other.end(), this->end());
+      __other.__destroy(__new_mid, __other.end());
+    }
+    else
+    {
+      const auto __new_mid = _CUDA_VSTD::swap_ranges(__other.begin(), __other.end(), this->begin());
+      __other.__uninitialized_move(__new_mid, this->end(), __other.end());
+      this->__destroy(__new_mid, this->end());
+    }
+  }
+
+  _LIBCUDACXX_TEMPLATE(class _Tp2 = _Tp)
+  _LIBCUDACXX_REQUIRES(_CCCL_TRAIT(is_swappable, _Tp2) _LIBCUDACXX_AND _CCCL_TRAIT(is_move_constructible, _Tp2))
+  _LIBCUDACXX_INLINE_VISIBILITY friend constexpr void swap(inplace_vector& __lhs, inplace_vector& __rhs) noexcept(
+    _Capacity == 0 || (_CCCL_TRAIT(is_nothrow_swappable, _Tp2) && _CCCL_TRAIT(is_nothrow_move_constructible, _Tp2)))
+  {
+    __lhs.swap(__rhs);
+  }
+
+  // inplace_vector.comparison
+  _CCCL_NODISCARD_FRIEND _LIBCUDACXX_INLINE_VISIBILITY constexpr bool
+  operator==(const inplace_vector& __lhs, const inplace_vector& __rhs) noexcept(
+    noexcept(_CUDA_VSTD::equal(__lhs.begin(), __lhs.end(), __rhs.begin(), __rhs.end())))
+  {
+    return _CUDA_VSTD::equal(__lhs.begin(), __lhs.end(), __rhs.begin(), __rhs.end());
+  }
+#  if _CCCL_STD_VER <= 2017
+  _CCCL_NODISCARD_FRIEND _LIBCUDACXX_INLINE_VISIBILITY constexpr bool
+  operator!=(const inplace_vector& __lhs, const inplace_vector& __rhs) noexcept(
+    noexcept(_CUDA_VSTD::equal(__lhs.begin(), __lhs.end(), __rhs.begin(), __rhs.end())))
+  {
+    return !_CUDA_VSTD::equal(__lhs.begin(), __lhs.end(), __rhs.begin(), __rhs.end());
+  }
+#  endif // _CCCL_STD_VER <= 2017
+
+#  ifndef _LIBCUDACXX_HAS_NO_SPACESHIP_OPERATOR
+  _CCCL_NODISCARD_FRIEND _LIBCUDACXX_INLINE_VISIBILITY constexpr __synth_three_way_result_t<_Tp>
+  operator<=>(const inplace_vector& __lhs, const inplace_vector& __rhs) noexcept(
+    noexcept(_CUDA_VSTD::lexicographical_compare(__lhs.begin(), __lhs.end(), __rhs.begin(), __rhs.end())))
+  {
+    return _CUDA_VSTD::lexicographical_compare(__lhs.begin(), __lhs.end(), __rhs.begin(), __rhs.end());
+  }
```

Comment:
I believe it's `lexicographical_compare_three_way` that should be called here. However, IIUC it hasn't been implemented in libcudacxx yet...

Reply:
You are absolutely right. It is here just for completeness as we currently cannot support `<=>` due to host device issues